spyder/plugins/editor/widgets/editor.py

```diff
@@ -2821,7 +2821,11 @@ def _run_cell_text(self, text, editor, cell_id, debug=False):
                 self.debug_cell_in_ipyclient.emit(*args)
             else:
                 self.run_cell_in_ipyclient.emit(*args)
-        editor.setFocus()
+        if self.focus_to_editor:
+            editor.setFocus()
+        else:
+            term = QApplication.focusWidget()

```

Comment:
@steff456 maybe a better name for this variable could be `console` (just to be clear that the focused widget is the console).